### PR TITLE
Travis cmake

### LINKED ISSUE #182: Use CMake instead of Scons

### PR DESCRIPTION
**Pull request by narekgharibyan**
@hendrikmuhs this fixes #182 

Just don't know how to be with `deb` packaging, also docs needs to be updated.

### CODE REVIEW

**Comment by narekgharibyan** [comment]
And looks I break coveralls :/

**Comment by hendrikmuhs** [comment]
Yes, coverage is an own mode or 'build type' as it is called in cmake.

I wonder if the cmake files already covers everything that scons had. Looks a bit to simple for me. ;-)

E.g. there are also some TPIE flags scons sets.

I will try your PR next week and give feedback. Lets see that we get everything right before the switch.

**Comment by hendrikmuhs** [comment]
Nice, works for me.

@narekgharibyan anything missing except the doc changes? Any chance to move the  CMakeLists.txt into the keyvi subfolder?

**Comment by narekgharibyan** [comment]
@hendrikmuhs Thanks!

remaining things are: 
 - doc update
 - removing Scons scripts
 - open question about Debian packaging: https://github.com/cliqz-oss/keyvi/blob/master/keyvi/deb/SConscript, is it fine to remove it ?

regarding to CMakeLists.txt location: I checked and still it's a bit tricky to import project into CLion, I prefer to keep it in the root, but that's not a mandatory.

**Comment by hendrikmuhs** [comment]
@narekgharibyan 

This debian packaging you mentioned is outdated, so just remove it, no problem. The real debian packaging stuff is in the keyvi-package repo. It needs adjustment: 

https://github.com/cliqz-oss/keyvi-package/blob/master/debian/rules 

... but this can be done afterwards IMO.

As for the CMakeLists: I do not understand it completely - still haven't tried CLion: You need it in root to have 1 project for both keyvi and pykeyvi? If the file is located under keyvi, you would not be able to see pykeyvi? Is that ritght? 

Actually that's how I work using eclipse/pycharm. I have 2 projects there: a keyvi and a pykeyvi project. As both have there own build system I also see them as 2 projects.

Anyway, as it is just 1 file, I do not insist on putting it into the keyvi subfolder although I believe it belongs there.

**Comment by narekgharibyan** [comment]
@hendrikmuhs exactly!

**Comment by narekgharibyan** [comment]
@hendrikmuhs 
The compile section in file https://github.com/cliqz-oss/keyvi/blob/master/keyvi/README.md#compile looks to me a bit incorrect, as we are not compiling C++ library, rather we compile 3rd party lib and create standalone executables and unit tests executable. What do you think ?

**Comment by hendrikmuhs** [comment]
@narekgharibyan Correct, in the very early days it was a library, but this isn't the case for a long time (not even sure if it ever was the case for 'keyvi' or the predecessor).

Anyway, should be rephrased, good catch.

File: README.md [hunk 1]
```diff
@@ -40,10 +40,6 @@ For the python extension pykeyvi check the [Readme](/pykeyvi/README.md) file in
   * [Crashcourse](/doc/usage/Crashcourse.md)
   * [Using pykeyvi with EMR (mrjob or pyspark)](/doc/usage/Using pykeyvi in EMR.md)  
 
-## Development
- 
-  * [Build/Packaging](/doc/development/Packaging keyvi.md)
```
Comment (hendrikmuhs) [inline README.md]: what about linking keyvi-package?

https://github.com/cliqz-oss/keyvi-package

Comment (narekgharibyan) [inline README.md]: Not sure.
Anyway now `keyvi-package` repo is a bit outdated. Let's revisit this later on.

File: keyvi/README.md [hunk 2]
```diff
@@ -23,11 +12,17 @@ For example on Ubuntu (14.04 to 16.04) should install all the dependencies you n
...
     brew install boost --c++11
     brew install snappy
     brew install lzlib
     brew install cmake
 
 Now you should be able to compile as explained above.
+
+## Build
+
+Use `cmake` to build keyvi executables along with unit tests.
```
Comment (hendrikmuhs) [inline keyvi/README.md]: can you add howto do debug and/or release builds?

Comment (narekgharibyan) [inline keyvi/README.md]: @hendrikmuhs please have a look

**Comment by hendrikmuhs** [comment]
LGTM!

(can you do a squash merge for less history polluting)

**Pull request merged**